Community/Tdarr_Plugin_TN10_SUBS.js

```diff
@@ -0,0 +1,193 @@
+// jshint esversion: 6
+const fs = require('fs');
+
+// Created by drpeppershaker with help from reddit user /u/jakejones48, lots of improvements made after looking at "Tdarr_Plugin_078d" by HaveAGitGat. 
+// Further modified by tehNiemer to allow language specification, overwriting of existing subtitle files, and properly recognizing and extracting forced subtitles.
+
+function details() {
+  return {
+    id: 'Tdarr_Plugin_TN10_SUBS',
+    Stage: 'Pre-processing',
+    Name: 'tehNiemer - Extract embedded subtitles and optionally remove them',
+    Type: 'Video',
+    Operation: 'Transcode',
+    Description: 'This plugin extracts embedded .srt subs in one pass inside Tdarr and will optionally remove them as well as all image based subtitles. This will not extract commentary files.\n\n ',
+    Version: '1.00',
+    Link: '',
+    Tags: 'pre-processing,subtitle only,ffmpeg,configurable',
+    Inputs: [{
+        name: 'extract_lang',
+        type: 'string',
+        defaultValue: 'eng',
+        inputUI: {
+          type: 'text',
+        },
+        tooltip: 'Specify language tag(s) here for the subtitle tracks you would like to extract.\\nLeave blank to extract all subtitle tracks.\\nMust follow ISO-639-2 3 letter format. https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes.\\nExample: \\neng\\nExample: \\neng,jpn'
+      },
+      {
+        name: 'remove_subs',
+        type: 'boolean',
+        defaultValue: 'false',
+        inputUI: {
+          type: 'dropdown',
+          options: [
+            'false',
+            'true',
+          ],
+        },
+        tooltip: 'Remove all subtitle streams from file after extrations.\\nThis will also remove image format subtitles that can not be extracted.'
+      },
+      {
+        name: 'overwrite_subs',
+        type: 'boolean',
+        defaultValue: 'false',
+        inputUI: {
+          type: 'dropdown',
+          options: [
+            'false',
+            'true',
+          ],
+        },
+        tooltip: 'Overwrite existing subtitle files on disk if they exist.'
+      },
+    ],
+  };
+}
+
+function plugin(file, librarySettings, inputs, otherArguments) {
+  // Must return this object at some point in the function else plugin will fail.
+  const response = {
+    processFile: true,
+    preset: '',
+    container: `.${file.container}`,
+    handBrakeMode: false,
+    FFmpegMode: true,
+    reQueueAfter: false,
+    infoLog: '',
+  };
+
+  // Check if file is a video. If it isn't then exit plugin.
+  if (file.fileMedium !== 'video') {
+    // eslint-disable-next-line no-console
+    response.infoLog += '☒File is not video \n';
+    response.processFile = false;
+    return response;
+  }
+
+  // Make sure file has subtitles.
+  let hasSubs = false;
+  for (let i = 0; i < file.ffProbeData.streams.length; i++) {
+    try {
+      if (
+        file.ffProbeData.streams[i].codec_type.toLowerCase() === 'subtitle'
+      ) {
+        hasSubs = true;
+      }
+    } catch (err) {}
+  }
```

Comment:
could be simplified to `const hasSubs = file.ffProbeData.streams.filter(x => x.codec_type === 'subtitle').length > 0;`

Reply:
Fixed with next pull